tests/jobs/auto_correlate_observations/unit/test_auto_correlate_observations.py

```diff
@@ -0,0 +1,239 @@
+import configsuite
+import itertools
+import numpy as np
+import pandas as pd
+import pytest
+import random
+
+from semeio.jobs import auto_correlate_observations
+
+
+class MockedMeasuredData(object):
+    def __init__(self, observations, responses):
+        self._data = self._build_data(observations, responses)
+
+    def _build_data(self, observations, responses):
+        def assert_equal_keys(a, b):
+            assert sorted(a.keys()) == sorted(b.keys())
+
+        columns = []
+        ens_size = None
+        assert_equal_keys(observations, responses)
+        for key in observations.keys():
+            assert_equal_keys(observations[key], responses[key])
+            for sub_key, sim in responses[key].items():
+                if ens_size is None:
+                    ens_size = len(sim)
+                assert ens_size == len(sim)
+                columns.append((key,) + tuple(sub_key))
+
+        index = ("OBS", "STD") + tuple(range(ens_size))
+
+        raw_data = np.array(
+            [
+                np.array([observations[col[0]][col[1:]][0] for col in columns]),
+                np.array([observations[col[0]][col[1:]][1] for col in columns]),
+            ]
+        )
+        simulated_raw_data = np.array([responses[col[0]][col[1:]] for col in columns]).T
+        raw_data = np.concatenate((raw_data, simulated_raw_data))
+
+        data = pd.DataFrame(
+            raw_data,
+            index=index,
+            columns=pd.MultiIndex.from_tuples(
+                columns, names=[None, "key_index", "data_index"]
+            ),
+        )
+
+        return data
+
+    @property
+    def data(self):
+        return self._data
+
+    def get_simulated_data(self):
+        return self._data.drop(index=["OBS", "STD"])
+
+    def remove_failed_realizations(self):
+        pass
+
+    def remove_inactive_observations(self):
+        pass
+
+    def filter_ensemble_std(self, _):
+        pass
+
+
+def assertHomogenClusters(config):
```

Comment:
Why camelCase? Also why this is called homogenCluster?

Reply:
camelCase due to old habit of following unittest syntax for assert functions... `homogenCluster` because it asserts that all members of each cluster comes from the same polynomial.